orc8r/cloud/go/obsidian/swagger/combine_test.go

```diff
@@ -0,0 +1,103 @@
+/*
+ Copyright 2020 The Magma Authors.
+
+ This source code is licensed under the BSD-style license found in the
+ LICENSE file in the root directory of this source tree.
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package swagger_test
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"magma/orc8r/cloud/go/obsidian/swagger"
+	"magma/orc8r/cloud/go/obsidian/swagger/protos"
+	"magma/orc8r/cloud/go/orc8r"
+	swagger_lib "magma/orc8r/cloud/go/swagger"
+	"magma/orc8r/cloud/go/test_utils"
+
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/yaml.v2"
+)
+
+func Test_GetCommonSpec(t *testing.T) {
+	specPath := "/etc/magma/configs/orc8r/swagger_specs"
+	commonSpecDir := "/etc/magma/configs/orc8r/swagger_specs/common"
+	commonSpecFilePath := filepath.Join(commonSpecDir, "swagger-common.yml")
+
+	os.RemoveAll(specPath)
+	defer os.RemoveAll(specPath)
+
+	err := os.MkdirAll(commonSpecDir, os.ModePerm)
+	assert.NoError(t, err)
+
+	commonTag := swagger_lib.TagDefinition{Name: "Tag Common"}
+	commonSpec := swagger_lib.Spec{Tags: []swagger_lib.TagDefinition{commonTag}}
+	yamlCommon := marshalToYAML(t, commonSpec)
+
+	err = ioutil.WriteFile(commonSpecFilePath, []byte(yamlCommon), 0644)
+	assert.NoError(t, err)
+
+	actual, err := swagger.GetCommonSpec()
+	assert.NoError(t, err)
+	assert.Equal(t, yamlCommon, actual)
+}
+
+func Test_GetCombinedSwaggerSpecs(t *testing.T) {
+	commonTag := swagger_lib.TagDefinition{Name: "Tag Common"}
+	commonSpec := swagger_lib.Spec{Tags: []swagger_lib.TagDefinition{commonTag}}
+	yamlCommon := marshalToYAML(t, commonSpec)
+
+	tags := []swagger_lib.TagDefinition{
+		{Name: "Tag 1"},
+		{Name: "Tag 2"},
+		{Name: "Tag 3"},
+	}
+	services := []string{"test_service1", "test_service2", "test_service3"}
+	assert.Equal(t, len(tags), len(services))
```

Comment:
I see, my reasoning for adding this check was so that it would be easier for any developer updating the test case in the future to think "oh there must be some reason why" and then resultingly check that we are doing a indexed based for loop which would break if the lengths were unequal. I can see how it's simple enough to not be needed though?